File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hashSet = set()
        
        for num in nums:
            hashSet.add(num)
            
        maxLen = 0
        
        for i in range(len(nums)):
            curr = nums[i]
            if curr - 1 not in hashSet:
                length = 0
                while (curr + length) in hashSet:
                    length += 1
                
                maxLen = max(maxLen, length)
        
        return maxLen

#Time Complexity: O(n)
#Space Complexity: O(n)
                    
                 
                
        
        
            
        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums == null || nums.length == 0){
            return 0;
        }
        
        int longest = 0;
        HashSet<Integer> set = new HashSet<>();
        
        for(int num: nums){
            set.add(num);
        }
        
        // System.out.println(set);
        
        for(int num : nums){
            if(!set.contains(num - 1)){
                
                int length = 0;
                while(set.contains(num + length)){
                    length += 1;
                }
                longest = Math.max(longest, length);
            }
        }
        
        return longest;
    }
}